core/src/main/java/org/neo4j/ogm/metadata/FieldInfo.java

```diff
@@ -160,17 +161,27 @@ public String property() {
     }
 
     public String relationship() {
-        if (this.containingClassInfo.relationshipFields().contains(this)) {
-            if (annotations != null) {
-                AnnotationInfo relationshipAnnotation = annotations.get(Relationship.class);
-                if (relationshipAnnotation != null) {
-                    return relationshipAnnotation
-                        .get(Relationship.TYPE, RelationshipUtils.inferRelationshipType(getName()));
-                }
+        if (relationship == null) {
```

Comment:
I would prefer the method following the double-checked-locking pattern as applied in the pair of `convertedType/computeConvertedType`, given the fact that you went for a volatile optional already. Thanks.